generators/app/templates/sources/main/helpers/analytics/analytics.service.ts

```diff
@@ -0,0 +1,76 @@
+import app from 'main.module';
+import {ILogger, LoggerService} from 'helpers/logger/logger';
+import {IApplicationConfig} from 'main.constants';
+
+interface IWindowWithAnalytics extends ng.IWindowService {
+  googleAnalytics: any;
+}
+
+/**
+ * Analytics service: insert Google Analytics library in the page.
+ */
+export class AnalyticsService {
+
+  private logger: ILogger;
+  private analyticsAreActive = false;
+
+  constructor(private $window: IWindowWithAnalytics,
+              private config: IApplicationConfig,
+              logger: LoggerService) {
+
+    this.logger = logger.getLogger('analyticsService');
+
+    this.init();
+  }
+
+  /**
+   * Tracks a page change in google analytics.
+   * @param {String} url The url of the new page.
+   */
+  trackPage (url: string) {
+    if (this.analyticsAreActive) {
+      let urlWithoutParams = url;
+      let split = url.split('?');
+      if (split.length > 1) {
+        urlWithoutParams = split[0];
+      }
+      this.$window.googleAnalytics('send', 'pageview', urlWithoutParams);
+    }
+  }
+
+  /**
+   * Sends a track event to google analytics.
+   * @param {String} category The category to be sent.
+   * @param {String} action The action to be sent.
+   * @param {String=} label The label to be sent.
+   */
+  trackEvent (category: string, action: string, label?: string) {
+    if (this.analyticsAreActive) {
+      this.$window.googleAnalytics('send', 'event', category, action, label);
+    }
+  }
+
+  private init(): void {
+    if (this.config.analyticsAccount !== null) {
+      let analyticsScriptUrl = '//www.google-analytics.com/analytics.js';
+      this.createGoogleAnalyticsObject(window, document, 'script', analyticsScriptUrl, 'googleAnalytics');
```

Comment:
Both done (although replacing an explicit field name with an obfuscated one does not sit right with me).